src/de/mod.rs

```diff
@@ -1,6 +1,1324 @@
 //! Serde `Deserializer` module
 //!
-//! # Difference between `$text` and `$value` special names
+//! Due to complexity of the XML standard and that fact that serde was developed
+//! with JSON in mind, not all serde concepts smoothly laid down on XML. This
+//! leads to that fact that some XML concepts are inexpressible in terms of serde
+//! derives and you may require implement deserialization manually.
+//!
+//! The most notable restriction is distinguish between _elements_ and _attributes_,
+//! no other serde format haven't such a conception.
+//!
+//! Due to that the mapping is performed in the best effort manner.
+//!
+//!
+//!
+//! Table of Contents
+//! =================
+//! - [Mapping XML to Rust types](#mapping-xml-to-rust-types)
+//!   - [Optional attributes and elements](#optional-attributes-and-elements)
+//!   - [Choices (`xs:choice` XML Schema type)](#choices-xschoice-xml-schema-type)
+//!   - [Sequences (`xs:all` and `xs:sequence` XML Schema types)](#sequences-xsall-and-xssequence-xml-schema-types)
+//! - [Composition Rules](#composition-rules)
+//! - [Difference between `$text` and `$value` special names](#difference-between-text-and-value-special-names)
+//!   - [`$text`](#text)
+//!   - [`$value`](#value)
+//!     - [Primitives and sequences of primitives](#primitives-and-sequences-of-primitives)
+//!     - [Structs and sequences of structs](#structs-and-sequences-of-structs)
+//!     - [Enums and sequences of enums](#enums-and-sequences-of-enums)
+//! - [Frequently Used Patterns](#frequently-used-patterns)
+//!   - [`<element>` lists](#element-lists)
+//!
+//!
+//!
+//! Mapping XML to Rust types
+//! =========================
+//!
+//! Type names are never considered when deserializing, so you could name your
+//! types as you wish. Other general rules:
+//! - `struct` field name could be represented in XML only as an attribute name
+//!   or an element name;
+//! - `enum` variant name could be represented in XML only as an attribute name
+//!   or an element name;
+//! - the unit struct, unit type `()` and unit enum variant can be deserialized
+//!   from any valid XML content:
+//!   - attribute and element names;
+//!   - attribute and element values;
+//!   - text or CDATA content (including mixed text + CDATA content).
+//!
+//! <div style="background:rgba(120,145,255,0.45);padding:0.75em;">
+//!
+//! NOTE: examples, marked with `FIXME:` does not work yet -- any PRs that fixes
+//! that are welcome! The message after marker is a test failure message.
+//! Also, all that tests are marked with an `ignore` option, although their
+//! compiles. This is by intention, because rustdoc marks such blocks with
+//! an exclamation mark unlike `no_run` blocks.
```

Comment:
```
//! NOTE: examples marked with `FIXME:` do not work properly yet -- any PRs which fixes
//! them are welcome! The test failure message is noted underneath each of them.
//! Also, some tests are marked with an `ignore` option although they compile fine.
//! This is by intention, because rustdoc marks such blocks with an exclamation
//! mark unlike `no_run` blocks.
```

Could you elaborate on the last sentence also, it is not clear what you mean.

Reply:
See the difference:
![difference-between-no_run-and-ignore](https://user-images.githubusercontent.com/450131/209440005-672d3742-ed6e-4436-9fa9-0cd4fc7e1e58.png)

Reply:
Replaced "exclamation mark" with "information icon"